coreneuron/CMakeLists.txt

```diff
@@ -260,6 +260,11 @@ target_include_directories(coreneuron SYSTEM
 target_include_directories(coreneuron SYSTEM
                            PRIVATE ${CORENEURON_PROJECT_SOURCE_DIR}/external/CLI11/include)
 
+# See: https://en.cppreference.com/w/cpp/filesystem#Notes
+if(CMAKE_CXX_COMPILER_IS_GCC AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1)
+  target_link_libraries(coreneuron stdc++fs)
+endif()
```

Comment:
note: as this is a compile-time decision here, this is not sufficient if we consider the use case of wheels:
* wheel is built with gcc > 9.1 and hence `stdc++fs` is not linked
* user is now using neuron wheel with gcc < 9.1. this results into link error